### PR TITLE
Container styles

### PR DESCRIPTION
**Pull request by RebeccaAnne**
Implements color related portions of #472.  Padding related changes to come in a future change.

### CODE REVIEW

File: source/shared/cpp/ObjectModel/HostConfig.h [hunk 1]
```diff
@@ -93,45 +93,54 @@ struct ImageSetConfig
...
-    static ContainerConfig Deserialize(const Json::Value& json, const ContainerConfig& defaultValue);
+    static FactSetConfig Deserialize(const Json::Value& json, const FactSetConfig& defaultValue);
 };
 
-struct AdaptiveCardConfig
+struct ContainerStyleDefinition
 {
-    SpacingDefinition padding = { 8, 8, 8, 8 };
     std::string backgroundColor = "#FFFFFFFF";
+    ColorsConfig foregroundColors;
```
Comment (nesalang) [inline source/shared/cpp/ObjectModel/HostConfig.h]: I think this should be "fontColors"?

Comment (dclaux) [reply source/shared/cpp/ObjectModel/HostConfig.h]: Per a conversation this morning we have agreed to rename fontColors into foregroundColors as these colors might apply to more than just fonts in the future.

Comment (nesalang) [reply source/shared/cpp/ObjectModel/HostConfig.h]: thanks

File: source/uwp/Renderer/lib/XamlBuilder.cpp [hunk 2]
```diff
@@ -308,7 +327,10 @@ namespace AdaptiveCards { namespace XamlCardRenderer
     }
 
     _Use_decl_annotations_
-    ComPtr<IUIElement> XamlBuilder::CreateRootCardElement(IAdaptiveCard* adaptiveCard, IPanel** childElementContainer)
+    ComPtr<IUIElement> XamlBuilder::CreateRootCardElement(
+        IAdaptiveCard* adaptiveCard,
```
Comment (riarenas) [inline source/uwp/Renderer/lib/XamlBuilder.cpp]: >  [](start = 36, length = 1)

[nit] extra space

File: source/uwp/Renderer/lib/AdaptiveCardConfig.cpp [hunk 3]
```diff
@@ -33,14 +33,16 @@ namespace AdaptiveCards { namespace XamlCardRenderer
     }
 
     _Use_decl_annotations_
-    HRESULT AdaptiveCardConfig::get_BackgroundColor(ABI::Windows::UI::Color* value)
+    HRESULT AdaptiveCardConfig::get_AllowCustomStyle(boolean * allowCustomStyle)
```
Comment (riarenas) [inline source/uwp/Renderer/lib/AdaptiveCardConfig.cpp]: [nit] spacing around *

File: source/uwp/Renderer/lib/AdaptiveCardConfig.cpp [hunk 4]
```diff
@@ -33,14 +33,16 @@ namespace AdaptiveCards { namespace XamlCardRenderer
...
     {
-        return GetColorFromString(m_sharedAdaptiveCardConfig.backgroundColor, value);
+        *allowCustomStyle = m_sharedAdaptiveCardConfig.allowCustomStyle;
+        return S_OK;
     }
-
+
```
Comment (riarenas) [inline source/uwp/Renderer/lib/AdaptiveCardConfig.cpp]: [nit] extra spaces

File: source/uwp/Renderer/lib/AdaptiveHostConfig.cpp [hunk 5]
```diff
@@ -87,13 +87,13 @@ namespace AdaptiveCards { namespace XamlCardRenderer
     }
 
     _Use_decl_annotations_
-    HRESULT  AdaptiveHostConfig::get_Colors(IAdaptiveColorsConfig** colorsConfig)
+    HRESULT AdaptiveHostConfig::get_ContainerStyles(ABI::AdaptiveCards::XamlCardRenderer::IAdaptiveContainerStylesDefinition ** value)
```
Comment (riarenas) [inline source/uwp/Renderer/lib/AdaptiveHostConfig.cpp]: [nit] spacing around *

File: source/uwp/Renderer/lib/AdaptiveHostConfig.cpp [hunk 6]
```diff
@@ -87,13 +87,13 @@ namespace AdaptiveCards { namespace XamlCardRenderer
...
     {
-        return MakeAndInitialize<AdaptiveColorsConfig>(colorsConfig, m_sharedHostConfig.colors);
+        return MakeAndInitialize<AdaptiveContainerStylesDefinition>(value, m_sharedHostConfig.containerStyles);
     }
 
     _Use_decl_annotations_
-    HRESULT  AdaptiveHostConfig::put_Colors(IAdaptiveColorsConfig* )
+    HRESULT AdaptiveHostConfig::put_ContainerStyles(ABI::AdaptiveCards::XamlCardRenderer::IAdaptiveContainerStylesDefinition * value)
```
Comment (riarenas) [inline source/uwp/Renderer/lib/AdaptiveHostConfig.cpp]: [nit] spacing around *

File: source/uwp/Renderer/lib/XamlBuilder.cpp (hunk 2 above)
Comment (RebeccaAnne) [reply source/uwp/Renderer/lib/XamlBuilder.cpp]: :)

---
In reply to: [140530587](https://github.com/Microsoft/AdaptiveCards/pull/648#discussion_r140530587) [](ancestors = 140530587)

**Comment by RebeccaAnne** [comment]
Moving to a new branch

**Pull request closed without merging**